bpf/sockops/bpf_sockops.h

```diff
@@ -24,7 +24,7 @@ struct sock_key {
 		union v6addr	dip6;
 	};
 	__u8 family;
-	__u8 pad7;
+	__u8 protocol;
```

Comment:
I think this change also requires an update to the userspace counterpart, `sockmap.Sockmapkey` See [the Smoke test logs](https://github.com/cilium/cilium/pull/12628/checks?check_run_id=908666472#step:8:399).